src/main/java/com/rtbhouse/kafka/workers/api/KafkaWorkers.java

```diff
@@ -136,7 +138,16 @@ public KafkaWorkers(
             WorkerTaskFactory<K, V> factory,
             WorkerPartitioner<K, V> partitioner,
             ShutdownCallback callback) {
-        this.workers = new KafkaWorkersImpl<>(config, factory, partitioner, callback);
+        this(config, factory, partitioner, new SimpleRecordWeigher<>(), callback);
+    }
+
+    public KafkaWorkers(
+            WorkersConfig config,
+            WorkerTaskFactory<K, V> factory,
+            WorkerPartitioner<K, V> partitioner,
+            Weigher<WorkerRecord<K, V>> recordWeigher,
```

Comment:
Weigher implementation could be passed as classpath to WorkersConfig. It would be quite analogous to serializers/deserializers passed to kafka consumers/producers via its configuration.

Reply:
Would you remove Weigher from KafkaWorkers constructor then?

Reply:
Yes, I would like to leave previous constructors' signatures.

Reply:
Done